barretenberg/cpp/src/barretenberg/commitment_schemes/ipa/ipa.hpp

```diff
@@ -183,10 +183,12 @@ template <typename Curve> class IPA {
      */
     static bool verify(const std::shared_ptr<VK>& vk,
                        const OpeningClaim<Curve>& opening_claim,
-                       const std::shared_ptr<BaseTranscript>& transcript)
+                       const std::shared_ptr<NativeTranscript>& transcript)
     {
-        auto poly_degree = static_cast<size_t>(transcript->template receive_from_prover<uint64_t>("IPA:poly_degree"));
-        const Fr generator_challenge = transcript->get_challenge("IPA:generator_challenge");
+        auto poly_degree = static_cast<uint32_t>(transcript->template receive_from_prover<typename Curve::BaseField>(
```

Comment:
If it always has to be `bb::fr` then we should just hard code that.

Reply:
When I wrote this I thought that something would break if we were to replace bb::fr with another field, but I don't recall now where that thought came from.

Reply:
We also can't hardcode bb::fr everywhere since we want field_ct in the circuit case, not bb::fr.